bin/get_data.sh

```diff
@@ -0,0 +1,127 @@
+fi
+if [ ! -f grib/${BASE}.sfc.grib ]; then
+    python bin/download_an_all.py $DATE $TIME sfc &
+fi
+if [ ! -f grib/${BASE}.pv.grib ]; then
+    python bin/download_an_all.py $DATE $TIME pv &
+fi
+if [ ! -f grib/${BASE}.tl.grib ]; then
+    python bin/download_an_all.py $DATE $TIME tl &
+fi
+wait
+
+if [ ! -f grib/${BASE}.ml.grib ]; then
+   echo	FATAL `date` Model level file is missing
+   exit
+fi
+if [ ! -f grib/${BASE}.sfc.grib ]; then
+   echo	FATAL `date` Surface file is missing
+   exit
+fi
+if [ ! -f grib/${BASE}.pv.grib ]; then
+   echo	FATAL `date` Potential Vorticity level file is missing
+   exit
+fi
+if [ ! -f grib/${BASE}.tl.grib ]; then
+   echo	FATAL `date` Potential Temperature level file is missing
+   exit
+fi
+
+cat grib/${BASE}.ml.grib grib/${BASE}.sfc.grib > ${GRIB}
+
+# convert grib to netCDF, set init time
+cdo -f nc4c copy grib/${BASE}.tl.grib $tlfile
+ncatted -a units,time,o,c,"${time_units}" $tlfile
+cdo -f nc4c copy grib/${BASE}.pv.grib $pvfile
+ncatted -a units,time,o,c,"${time_units}" $pvfile
+cdo -f nc4c copy ${GRIB} $mlfile
+ncatted -a units,time,o,c,"${time_units}" $mlfile
+
+# Add pressure and geopotential height to model levels file
+bin/add_pressure_gph.sh input=$mlfile pressure_units=Pa gph_units="m^2s^-2"
+
+# Add ancillary information
+python bin/add_ancillary.py $mlfile --pv --theta --tropopause --n2 #--eqlat nan values cause issues for now due to no 180° coverage
+
+# separate sfc from ml variables
+ncks -7 -L 7 -C -O -x -vlev_2,n2,clwc,u,q,t,pressure,zh,cc,w,v,ciwc,pt,pv,mod_pv,o3,d $mlfile $sfcfile
+ncatted -O -a standard_name,msl,o,c,air_pressure_at_sea_level $sfcfile
+ncks -C -O -vtime,lev_2,lon,lat,n2,clwc,u,q,t,pressure,zh,cc,w,v,ciwc,pt,pv,mod_pv,o3,d,hyai,hyam,hybi,hybm,sp,lnsp $mlfile $tmpfile
+mv $tmpfile $mlfile
+
+# interpolate to different grids
+echo "Creating pressure level file..."
+cdo ml2pl,85000,50000,40000,30000,20000,15000,12000,10000,8000,6500,5000,4000,3000,2000,1000,500,100 $mlfile $plfile
+ncatted -O -a standard_name,plev,o,c,atmosphere_pressure_coordinate $plfile
+ncap2 -s "plev/=100;plev@units=\"hPa\"" $plfile $plfile-tmp
+mv $plfile-tmp $plfile
+ncks -7 -L 7 -C -O -x -v lev,sp,lnsp,nhyi,nhym,hyai,hyam,hybi,hybm $plfile $plfile
+
+echo "Creating potential temperature level file..."
+python bin/interpolate_missing_variables.py $mlfile $tlfile pt
+python bin/rename_standard.py $mlfile $tlfile
+ncatted -O -a standard_name,lev,o,c,atmosphere_potential_temperature_coordinate $tlfile
+ncatted -O -a standard_name,pv,o,c,ertel_potential_vorticity $tlfile
+ncks -O -7 -L 7 $tlfile $tlfile
+
+echo "Creating potential vorticity level file..."
+python bin/interpolate_missing_variables.py $mlfile $pvfile pv
+python bin/rename_standard.py $mlfile $pvfile
+ncatted -O -a standard_name,lev,o,c,atmosphere_ertel_potential_vorticity_coordinate $pvfile
+ncatted -O -a standard_name,pt,o,c,air_potential_temperature $plfile
+ncatted -O -a units,lev,o,c,"kelvin * meter ** 2 / kilogram / second" $pvfile
```

Comment:
Is this correct? we typically deal in microkelvin * ....
the values in MSS level drop-down should be 2/4 or so. The file I downloaded with your script has a level of "2000", but should be (probably?) 2e-6 in the units given here...

Reply:
Using grib_dump, I get the following information, which is consistent to my understanding:

```
[...]
  # Potential vorticity surface 10-9 K m2 kg-1 s-1 (grib1/local/ecmf/3.table , grib1/3.table)  
  indicatorOfTypeOfLevel = 117;
  level = 2000;
[...]
```
Where do you got "your" units from? Maybe there really is a bug in the ECMWF documentation?

I would specify this as `mPVU`.

Reply:
You are right, it is most likely wrong.
It seems the 2000 is in mPVU, not PVU, and as such in nanokelvin, not kelvin.
I divided it by 1000 to convert it to PVU and corrected the units attribute.

Reply:
In general PV is not in PVU (microkelvin) but kelvin https://apps.ecmwf.int/codes/grib/param-db?id=60
Except for the levels which are also not in PVU but in nanokelvin. Generally this was confusing me.